src/ORM/Marshaller.php

```diff
@@ -105,6 +105,18 @@ public function one(array $data, array $options = [])
         $propertyMap = $this->_buildPropertyMap($options);
 
         $schema = $this->_table->schema();
+        $primaryKey = $schema->primaryKey();
+
+        if (array_intersect($primaryKey, array_keys($data)) == $primaryKey) {
+            $record = $this->_table->find('all');
+            foreach ($primaryKey as $pkey) {
+                $record->where(["$pkey" => $data[$pkey] ]);
```

Comment:
You're not adding joins, but someone somewhere will use callbacks or override find() to add joins. We've had a number of similar issues reported when we thought we were 'safe'.